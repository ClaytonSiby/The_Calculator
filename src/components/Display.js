import PropTypes from 'prop-types';
import styles from '../assets/styles.module.css';
import display from '../assets/display.module.css';

const Display = ({ total, next, operation }) => (
  <div className={`${styles.flexContainer} ${display.displayComp}`}>
    <p>
      <span>{ total || '0' }</span>
      <br />
      <span>
        {' '}
        {operation || null}
        {' '}
        { next }
      </span>
    </p>
  </div>
);

Display.propTypes = {
  total: PropTypes.string.isRequired,
  next: PropTypes.string.isRequired,
  operation: PropTypes.string.isRequired,
};

export default Display;
