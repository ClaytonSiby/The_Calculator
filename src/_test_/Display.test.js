import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Display from '../components/Display';

describe('Display component', () => {
  test('should render successfully to the DOM', () => {
    render(<Display />);
  });

  test('should render correct', () => {
    const tree = renderer.create(<Display />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
