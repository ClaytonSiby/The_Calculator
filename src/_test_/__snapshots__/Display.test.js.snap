// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component Display renders correct 1`] = `
<div
  className="flexContainer displayComp"
>
  <p>
    <span>
      0
    </span>
    <br />
    <span>
       
       
      
    </span>
  </p>
</div>
`;

exports[`Display component should render correct 1`] = `
<div
  className="flexContainer displayComp"
>
  <p>
    <span>
      0
    </span>
    <br />
    <span>
       
       
      
    </span>
  </p>
</div>
`;
